pkg/apis/cosigned/v1alpha1/clusterimagepolicy_types.go

```diff
@@ -49,9 +36,81 @@ func (*ClusterImagePolicy) GetGroupVersionKind() schema.GroupVersionKind {
 	return SchemeGroupVersion.WithKind("ClusterImagePolicy")
 }
 
-// ClusterImagePolicySpec holds the desired state of the ClusterImagePolicy (from the client).
+// ClusterImagePolicySpec defines a list of images that should be verified
 type ClusterImagePolicySpec struct {
-	// TODO(#1417): Flesh out the specification from the API spec.
+	Images []ImagePattern `json:"images"`
+}
+
+// ImagePattern defines a pattern and its associated authorties
+// If multiple patterns match a particular image, then ALL of
+// those authorities must be satisfied for the image to be admitted.
+type ImagePattern struct {
+	Glob        string      `json:"glob"`
+	Regex       string      `json:"regex"`
+	Authorities []Authority `json:"authorities"`
+}
+
+// The authorities block defines the rules for discovering and
+// validating signatures.  Signatures are
+// cryptographically verified using one of the "key" or "keyless"
+// fields.
+// When multiple authorities are specified, any of them may be used
+// to source the valid signature we are looking for to admit an
+// image.
+
+type Authority struct {
+	// +optional
+	Key *KeyRef `json:"key,omitempty"`
+	// +optional
+	Keyless *KeylessRef `json:"keyless,omitempty"`
+	// +optional
+	Sources *[]Source `json:"source,omitempty"`
+	// +optional
+	CTLog *TLog `json:"ctlog,omitempty"`
+}
+
+// This references a public verification key stored in
+// a secret in the cosign-system namespace.
+// A KeyRef must specify only one of SecretRef, Data or KMS
+type KeyRef struct {
+	// +optional
+	SecretRef *v1.SecretReference `json:"secretref,omitempty"`
```

Comment:
This does allow for the namespace to be specified. I personally like this flexibility, and we can validate if a particular instance does not allow for per-namespace secrets. Just jotting it down, because in the api working group we had talked about the problem of having to manage secrets in many namespaces and decided that the way around that would be restrict that they live in a single namespace to reduce management headaches. This allows us to do both 👍